packages/next/src/server/app-render/dynamic-rendering.ts

```diff
@@ -145,18 +158,25 @@ export function trackDynamicFetch(
 ) {
   if (store.prerenderState) {
     assertPostpone()
-    store.prerenderState.hasDynamic = true
-    React.unstable_postpone(createPostponeReason(expression, store.urlPathname))
+    postponeWithTracking(store.prerenderState, expression, store.urlPathname)
   }
 }
 
-function createPostponeReason(expression: string, urlPathname: string) {
-  const pathname = getPathname(urlPathname) // remove queries such like `_rsc` for flight
-  return (
+function postponeWithTracking(
```

Comment:
would it make sense for `postponeWithTracking` to `assertPostpone`? I worry about having to remember to do both in all the relevant places